src/main/java/com/bookshop/controller/BookController.java

```diff
@@ -18,41 +25,203 @@
+    private static final String CODE_201 = "201";
+    private static final String CREATED_BOOK_DESCRIPTION = "Book created successfully";
+    private static final String CODE_204 = "204";
+    private static final String DELETED_BOOK_DESCRIPTION = "Book deleted successfully";
+    private static final String CODE_400 = "400";
+    private static final String CODE_400_DESCRIPTION = "Incorrect data was provided in the body";
+    private static final String CODE_400_EXAMPLE =
+            """
+            {
+                "timestamp": "2023-09-08T02:34:40.8612831",
+                "status": "BAD_REQUEST",
+                "errors": [
+                    "title can't be null, should be set"
+                ]
+            }
+            """;
+    private static final String CODE_404 = "404";
+    private static final String CODE_404_DESCRIPTION =
+            "Book with such id doesn't exists or was previously deleted";
+    private static final String CODE_404_EXAMPLE =
+            """          
+                {
+                    "timestamp": "*time of query*",
+                    "status": "NOT_FOUND",
+                    "errors": "Can't find book by id: {id}"
+                }
+            """;
+    private static final String CODE_500 = "500";
+    private static final String CODE_500_DESCRIPTION = "Internal server error";
+    private static final String CODE_500_EXAMPLE =
+            """
+                {
+                    "timestamp": "*time of query*",
+                    "status": "BAD_REQUEST",
+                    "errors": [
+                        "*problem description*"
+                    ]
+                }
+            """;
+    private static final String BOOK_DTO_EXAMPLE =
+            """
+                {
+                  "id": 0,
+                  "title": "string",
+                  "author": "string",
+                  "isbn": "978-3-16-148410-1",
+                  "price": 0,
+                  "description": "string",
+                  "coverImage": "string"
+                }
+            """;
+    private static final String MEDIA_TYPE = "application/json";
     private final BookService bookService;
 
     @GetMapping
-    public List<BookDto> getAll() {
-        return bookService.findAll();
+    @Operation(summary = "Get list of all books",
+            description = "Returns a list of books based on the provided paging information")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = CODE_200, description = GET_LIST_DESCRIPTION,
+            content = {@Content(mediaType = MEDIA_TYPE,
+                    examples = {@ExampleObject(value = BOOK_DTO_EXAMPLE)}
+                    )}
+            ),
+        @ApiResponse(responseCode = CODE_500, description = CODE_500_DESCRIPTION,
+            content = {@Content(mediaType = MEDIA_TYPE,
+                    examples = {@ExampleObject(value = CODE_500_EXAMPLE)}
+                    )}
+            )
+    })
```

Comment:
It looks impressive! What advantages have you got by using this approach instead of default swagger implementation?

Reply:
It is more accurate and allows me to show examples of the error messages